### ISSUE TITLE
PSA: examples/psa-example-register1.p4

### ISSUE THREAD

**Issue by jnfoster** [p4lang/p4-spec#437]
The program [psa-example-register1.p4](https://github.com/p4lang/p4-spec/blob/master/p4-16/psa/examples/psa-example-register1.p4#L112) store structs in registers. However, the current compiler front-end (i.e., `p4test`) rejects them with the following error:
> functions or methods returning structures are not supported on this target

**Comment by jafingerhut** [comment]
That is true.  I wrote psa-example-register1.p4 first, hoping it would work, and when realizing that it does not, I left it there as a reminder that it does not work, created the issue below in hopes that it will work in the future, and created psa-example-register2.p4 that is identical in functionality, but works around the inability to put a struct inside a Register, by packing/unpacking a bit vector "manually".

https://github.com/p4lang/p4-spec/issues/383

I think I may also have created an issue for the p4c repository as well, but not sure if I can easily find it again, if I did.
